controllers/ironicneutronagent_controller.go

```diff
@@ -744,35 +757,40 @@ func (r *IronicNeutronAgentReconciler) generateServiceConfigMaps(
 		return err
 	}
 
+	transportURL, err := r.getTransportURL(ctx, h, instance)
+	if err != nil {
+		return err
+	}
+
+	ospSecret, _, err := secret.GetSecret(ctx, h, instance.Spec.Secret, instance.Namespace)
+	if err != nil {
+		return err
+	}
+
 	templateParameters := make(map[string]interface{})
 	templateParameters["ServiceUser"] = instance.Spec.ServiceUser
 	templateParameters["KeystoneInternalURL"] = keystoneInternalURL
 	templateParameters["KeystonePublicURL"] = keystonePublicURL
+	templateParameters["TransportURL"] = transportURL
+
+	// Other OpenStack services
+	servicePassword := string(ospSecret.Data[instance.Spec.PasswordSelectors.Service])
+	templateParameters["ServicePassword"] = servicePassword
+	templateParameters["keystone_authtoken"] = servicePassword
+	templateParameters["service_catalog"] = servicePassword
+	templateParameters["ironic"] = servicePassword
 
 	cms := []util.Template{
-		// Scripts ConfigMap
-		{
-			Name:         fmt.Sprintf("%s-scripts", instance.Name),
-			Namespace:    instance.Namespace,
-			Type:         util.TemplateTypeScripts,
-			InstanceType: instance.Kind,
-			AdditionalTemplate: map[string]string{
-				"common.sh": "/common/bin/common.sh",
-			},
-			Labels: cmLabels,
-		},
-		// Custom ConfigMap
 		{
-			Name:          fmt.Sprintf("%s-config-data", instance.Name),
+			Name:          fmt.Sprintf("%s-config", instance.Name),
```

Comment:
This is a bit of nit picking, but can we keep the name `%s-config-data`?
This will re-use, owerwrite, the existing secret, instead of creating a new one. This means we don't leave old data behind.

NOTE: You wil also have to update the `SecretName` in `pkg/ironicneutronagent/volumes.go`.


Bonus, if we can clean up the `ironic-ironic-neutron-agent-scripts` secret as well if it exists? (This can be a separate PR)